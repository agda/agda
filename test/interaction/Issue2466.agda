-- Andreas, 2017-03-21, issue #2466

-- The unifier should not turn user-written variable patterns into
-- dot patterns.

-- {-# OPTIONS -v reify.implicit:100 -v interaction.case:100 #-}
-- {-# OPTIONS -v tc.lhs.unify:40 #-}

postulate
  A B : Set

module Explicit where

  data D : A → B → Set where
    c : ∀ p {p'} x → D p' x → D p x

  test : ∀ p x → D p x → D p x
  test .p _ (c p x pp) = {!p'!}
    where
    y = x

  -- Expected: test .p _ (c p {p'} x pp) = ?

module Implicit where

  data D : A → B → Set where
    c : ∀ {p p' x} → D p' x → D p x

  test : ∀ p {x} → D p x → D p x
  test .p (c {p} {x = x} pp) = {!p'!}
    where
    y = x

  -- Expected: test .p (c {p} {p'} {x = x} pp) = ?
