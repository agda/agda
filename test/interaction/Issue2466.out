(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : D p x ?1 : D p x " nil)
((last . 1) . (agda2-goals-action '(0 1)))
((last . 2) . (agda2-make-case-action '("test .p _ (c p {p'} x pp) = ?")))
((last . 1) . (agda2-goals-action '(0 1)))
((last . 2) . (agda2-make-case-action '("test .p (c {p} {p'} {x = x} pp) = ?")))
((last . 1) . (agda2-goals-action '(0 1)))
